### ISSUE TITLE
Size gets stripped when using uploadBytes

### ISSUE THREAD

**Issue by SimplyLinn** [firebase/firebase-js-sdk#5744]
### [REQUIRED] Describe your environment

  * Operating System version: MacOS 12.0.1
  * Browser version: Chrome, Version 95.0.4638.69 (Official Build) (x86_64)
  * Firebase SDK version: @firebase/storage@0.8.7, firebase@9.4.1
  * Firebase Product: storage

#### Steps to reproduce:
* Create a create- or write-rule in firebase storage that limits uploaded file size
* Use uploadBytes to upload a file

[At this line](https://github.com/firebase/firebase-js-sdk/blob/ce39a1a07e8710e43cc66d9e7db882f185211a9a/packages/storage/src/implementation/requests.ts#L341) the size gets injected via `metadataForUpload_()`, but then, on the line below, it gets stripped due to the [mappings defining size as a non-writeable metadata field](https://github.com/firebase/firebase-js-sdk/blob/ce39a1a07e8710e43cc66d9e7db882f185211a9a/packages/storage/src/implementation/metadata.ts#L98) (false is the default value)

This breaks any uploads where one wants to limit the uploaded size via storage rules.

#### Relevant Code:

```ts
  const handleSubmit = useCallback(
    (ev: React.FormEvent<HTMLFormElement>) => {
      ev.preventDefault();
      const data = new FormData(ev.currentTarget);
      let image = data.get('image');
      if (!(image instanceof File) || image.size === 0) image = null;
      const file = ref(storage, `profile_pictures/user/original`);
      let promise: Promise<unknown>;
      if (image) {
        promise = uploadBytes(file, image);
      } else {
        promise = deleteObject(file);
      }
      promise
        .then(() => {
          console.log(success);
        })
        .catch((err) => console.error(err));
    },
    [],
  );
```

**Comment by jbalidiong** [comment]
Hi @SimplyLinn, thanks for reaching out to us. Could you share a few more details, like logs you are receiving when you're uploading the image?

**Comment by SimplyLinn** [comment]
@jbalidiong 
I went to create a minimum example of solving this. And was unable to reproduce, and I just tried the ruleset that didn't work a few days ago, and that also suddenly worked. So while I cannot produce any logs, I do remember it was the standard 403 missing write/create permissions.

While I still think it's weird that the size gets injected into the metadata and then stripped in the next line, it's seems actually unrelated to the issue I had. I just stumbled upon that fact while I was furiously troubleshooting our upload issues last week and ended up spotting the behavior of stripping of the size metadata, and it seemed to play into exactly my issue. Since my issue disappeared when I commented out the size part of the rules.

**Comment by schmidt-sebastian** [comment]
@SimplyLinn I am not able to tell from your response if you are still seeing this issue with your new code. If not, do you know what you changed? @jbalidiong created a small repro and was not able to reproduce this error. In our example we saw that the backend enforced the rules correctly.

**Comment by SimplyLinn** [comment]
@schmidt-sebastian no, the enforcement is no longer an issue. I either had a conflicting rule that caused the issue and got removed between me doing the testing (which would be weird, because I can swear I commented out the size-rules and it worked, and then stopped when I put them back), or some intermittent/temporary thing just caused the issue at the time of troubleshooting.

The code I highlighted I still think is doing something weird, but it's now unrelated to the issue of uploading files with size restriction. I think the library should either not inject the size-property in the first place, or it should not strip the size property once it's there. This is, however, now unrelated from the issue that caused me to start digging, but it is something I feel is an odd behavior.
